Integrations/Autodesk/maya/scripts/unityCommands.mel

```diff
@@ -633,13 +668,277 @@ proc setupNewExportSet(string $exportPath, int $exportAnimOnly, string $selected
+        error ("Unity FBX Export Set Creation: Nothing selected");
+        return;
+    }
+
+    string $namespace = `textField -q -text $namespaceField`;
+    if ($namespace == ""){
+        $namespace = ":";
+    }
+
+    string $modelPath = "";
+    string $modelFilename = "";
+    if($modelPathField != 0){
+        $modelPath = `textField -q -text $modelPathField`;
+        $modelFilename = `textField -q -text $modelFileField`;
+        if ($modelFilename != "" && !endsWith($modelFilename, ".fbx")){
+            $modelFilename = $modelFilename + ".fbx";
+        }
+    }
+
+    string $animPath = "";
+    string $animFilename = "";
+    if($animPathField != 0){
+        $animPath = `textField -q -text $animPathField`;
+        $animFilename = `textField -q -text $animFileField`;
+        if ($animFilename != "" && !endsWith($animFilename, ".fbx")){
+            $animFilename = $animFilename + ".fbx";
+        }
+    }
+
+    // make sure all necessary variables are set
+    if ($modelFilename == "" && $animFilename == ""){
+        error ("Unity FBX Export Set Creation: Missing filename for export.");
+        return;
+    }
+    
+    if ($modelPath == "" && $animPath == ""){
+        error ("Unity FBX Export Set Creation: Missing filepath for export.");
+        return;
+    }
+    
+    setupNewExportSet(
+        $namespace,
+        $modelPath,
+        $modelFilename,
+        $animPath,
+        $animFilename,
+        $origSelection);
+    
+    deleteUI -window $window;
+}
+
+global proc unityOpenFileDialog(string $textField)
+{
+    string $currentPath = `textField -q -text $textField`;
+    
+    string $exportPaths[] = `fileDialog2 -ds 2 -cap "FBX Export Selection" -dir $currentPath -fm 3`;
+    if(size($exportPaths)<=0){
+        return;
+    }
+    string $exportFilePath = $exportPaths[0];
+    
+    textField -e -text $exportFilePath $textField;
+}
+
+proc string createFilePathField(string $label, string $parent, int $labelSize, int $textFieldSize){
+    rowLayout
+        -numberOfColumns 3
+        -columnWidth3 $labelSize $textFieldSize 50
+        -columnAlign3 "right" "left" "left"
+        -p $parent;
+        
+    string $unityProject = `optionVar -q "UnityProject"`;
+    $unityProject = $unityProject + "/Assets";
+    
+    text -label $label;
+    string $textField = `textField -width $textFieldSize -text $unityProject`;
+    button -label "..." -recomputeSize false -height 15 -command ("unityOpenFileDialog " + $textField);
+    
+    return $textField;
+}
+
+proc string createTextFieldWithLabel(string $label, string $parent, int $labelSize, int $textFieldSize)
+{
+    rowLayout 
+        -numberOfColumns 2 
+        -columnWidth2 $labelSize $textFieldSize
+        -columnAlign2 "right" "left"
+        -p $parent;
+    
+    text -label $label;
+    string $textField = `textField -width $textFieldSize`;
+    return $textField;
+}
+
+proc string getCommonNamespace(string $origSelection[]){
```

Comment:
So if I get this right, there's a bunch of objects and we want to get the "root-est" namespace?

Reply:
yes, if you select a bunch of objects and most of them are in the same namespace, then take that one. So if you have:

test1:test2:sphere
test1:test3:cube
test1:test2:cylinder

If should find `test1` because all objects have that namespace in common. But if you have:

test1:test2:sphere
test1:test2:cube
test1:test2:cylinder

Then it will take `test1:test2`.

I'll add this as a comment to the function.